decidim-admin/app/permissions/decidim/admin/permissions.rb

```diff
@@ -64,6 +64,12 @@ def permissions
             permission_action.action == :destroy ? allow_destroy_taxonomy? : allow!
           end
 
+          unless current_participatory_space_is_initiative?
+            allow! if permission_action.subject == :soft_delete
+            allow! if permission_action.subject == :deleted_components
+            allow! if permission_action.subject == :restore_component
+          end
```

Comment:
```suggestion
          allow! if soft_delete_action? unless current_participatory_space_is_initiative?
...

          def soft_delete_action?
            if permission_action.subject == :component && permission_action.action.in?(:manage_trash, :restore, etc)
          end
```

Here, it would be better to maintain the meaning of "subject" and action, the subject is the object the action, what you are going to do with it.
So better check if the particular action is allowed for a particular subject (a component or a participatory space for instance)